roles/openshift_openstack/defaults/main.yml

```diff
@@ -46,13 +46,15 @@ openshift_openstack_container_storage_setup:
 openshift_openstack_dns_records_add: []
 
 openshift_openstack_app_subdomain: 'apps'
-openshift_openstack_public_hostname_suffix: "{{ openshift_openstack_clusterid|trim }}"
-openshift_openstack_private_hostname_suffix: "{{ openshift_openstack_clusterid|trim }}"
+openshift_openstack_public_hostname_suffix: ''
+openshift_openstack_private_hostname_suffix: ''
 openshift_openstack_public_dns_domain: 'example.com'
 
 openshift_openstack_private_dns_domain: "{{ openshift_openstack_public_dns_domain }}"
-openshift_openstack_full_public_dns_domain: "{{ (openshift_openstack_public_hostname_suffix == '') | ternary(openshift_openstack_public_dns_domain, openshift_openstack_public_hostname_suffix + '.' + openshift_openstack_public_dns_domain) }}"
-openshift_openstack_full_private_dns_domain: "{{ (openshift_openstack_private_hostname_suffix == '') | ternary(openshift_openstack_private_dns_domain, openshift_openstack_private_hostname_suffix + '.' + openshift_openstack_private_dns_domain) }}"
+openshift_openstack_public_subdomain: "{{ openshift_openstack_clusterid|trim + '.' + openshift_openstack_public_dns_domain }}"
+openshift_openstack_private_subdomain: "{{ openshift_openstack_clusterid|trim + '.' + openshift_openstack_private_dns_domain }}"
+openshift_openstack_full_public_dns_domain: "{{ (openshift_openstack_public_hostname_suffix|trim == '') | ternary(openshift_openstack_public_subdomain, openshift_openstack_public_hostname_suffix + '.' + openshift_openstack_public_subdomain) }}"
+openshift_openstack_full_private_dns_domain: "{{ (openshift_openstack_private_hostname_suffix|trim == '') | ternary(openshift_openstack_private_subdomain, openshift_openstack_private_hostname_suffix + '.' + openshift_openstack_private_subdomain) }}"
```

Comment:
@tomassedovic I think subdomains give more flexibility than just hostname suffixes. You cannot alter private domain fully with the latter it seems.